forc-plugins/forc-node/src/chain_config.rs

```diff
@@ -0,0 +1,577 @@
+    /// directory from: https://github.com/FuelLabs/chain-configuration/.
+    pub async fn download_config(&self, conf: &ChainConfig) -> anyhow::Result<()> {
+        let folder_name = match conf {
+            ChainConfig::Local => LOCAL_CONFIG_FOLDER_NAME,
+            ChainConfig::Testnet => TESTNET_CONFIG_FOLDER_NAME,
+            ChainConfig::Ignition => IGNITION_CONFIG_FOLDER_NAME,
+        };
+
+        let api_endpoint = format!(
+            "https://api.github.com/repos/FuelLabs/{}/contents/{}",
+            CHAIN_CONFIG_REPO_NAME, folder_name,
+        );
+
+        let contents = self.fetch_folder_contents(&api_endpoint).await?;
+
+        // Create config directory if it doesn't exist
+        let config_dir = user_forc_directory().join(CONFIG_FOLDER);
+        let target_dir = config_dir.join(folder_name);
+        fs::create_dir_all(&target_dir)?;
+
+        // Download each file
+        for item in contents {
+            if item.content_type == "file" {
+                if let Some(download_url) = item.download_url {
+                    let file_path = target_dir.join(&item.name);
+
+                    let response = self.client.get(&download_url).send().await?;
+
+                    if !response.status().is_success() {
+                        bail!("Failed to download file: {}", item.name);
+                    }
+
+                    let content = response.bytes().await?;
+                    fs::write(file_path, content)?;
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Helper function to fetch folder contents from github.
+    async fn fetch_folder_contents(&self, url: &str) -> anyhow::Result<Vec<GithubContentDetails>> {
+        let response = self
+            .client
+            .get(url)
+            .header("User-Agent", "forc-node")
+            .send()
+            .await?;
+
+        if !response.status().is_success() {
+            bail!("failed to fetch contents from github");
+        }
+
+        Ok(response.json().await?)
+    }
+}
+
+/// Local configuration is validated based on its existance. Meaning that if
+/// the configuration exists in user's local it is validated. If it is missing
+/// the configuration files are fetched from remote.
+async fn validate_local_chainconfig(fetcher: &ConfigFetcher) -> anyhow::Result<()> {
+    let user_conf_dir = user_forc_directory().join(CONFIG_FOLDER);
+    let local_conf_dir = user_conf_dir.join(LOCAL_CONFIG_FOLDER_NAME);
+    if !local_conf_dir.exists() {
+        println_warning(&format!(
+            "Local node configuration files are missing at {}",
+            local_conf_dir.display()
+        ));
+        // Ask user if they want to update the chain config.
+        let update = ask_user_yes_no_question("Would you like to download network configuration?")?;
+        if update {
+            fetcher.download_config(&ChainConfig::Local).await?;
+        } else {
+            bail!(
```

Comment:
For some reason these errors aren't being formatted properly.
![image](https://github.com/user-attachments/assets/7ee767fe-dfbc-4e1b-a379-cef9e696cd8d)